xbmc/cores/AudioEngine/Sinks/AESinkPULSE.cpp

```diff
@@ -315,6 +318,24 @@ static void SinkChangedCallback(pa_context *c, pa_subscription_event_type_t t, u
   }
 }
 
+struct ServerInfoStruct
+{
+  std::mutex m_mutex;
+  std::condition_variable m_cv;
+  bool m_isInfoSet{false};
+
+  bool m_isPipeWireServer{}; ///< true if server name contains "PipeWire" in any casing
+};
+
+static void ServerInfoCallback(pa_context* c, const pa_server_info* i, void* userdata)
+{
+  auto serverInfo = reinterpret_cast<ServerInfoStruct*>(userdata);
+  std::unique_lock l(serverInfo->m_mutex);
+  serverInfo->m_isPipeWireServer = StringUtils::Contains(i->server_name, "pipewire", true);
+  serverInfo->m_isInfoSet = true;
+  serverInfo->m_cv.notify_one();
```

Comment:
Any reason you do not use pa mainloop here?

Reply:
No, just ignorance 😉 I'll change that.

Reply:
You seem to have rebased on Omega.

Reply:
What I really don't like in this change is, that we now check for pipewire inside pulseaudio. This check and ifdef check should not belong into PulseAudio Sink. Also users having built with pipewire cannot use pulseaudio anymore and that would restore the unfortunate situation we had before ... that flatpak would output nothing as it would - cause of kodi building with pipewire AND pulseaudio support - default to pipewire.

The idea of reverting to pulse by default from my side was to give pipewire people a choice while not breaking existing setups.

Reply:
> You seem to have rebased on Omega.

Nope? Do you think that because I picked your commit from Omega?

> What I really don't like in this change is, that we now check for pipewire inside pulseaudio. This check and ifdef check should not belong into PulseAudio Sink.

Yes, this is a bit unfortunate but as PipeWire can't give this information we have to pull it from PulseAudio (https://gitlab.freedesktop.org/pipewire/pipewire/-/issues/1835). We could also pull PulseAudio into the PipeWire implementation to perform the check there but is that better?

> Also users having built with pipewire cannot use pulseaudio anymore and that would restore the unfortunate situation we had before ... that flatpak would output nothing as it would - cause of kodi building with pipewire AND pulseaudio support - default to pipewire.

It only defaults to PipeWire if PipeWire is driving the audio. If PulseAudio is running then PulseAudio is used. I'm talking about the real PulseAudio here, not the compatibility provided by PipeWire. With `--audio-backend=pulseaudio` the user can force the usage of the PulseAudio compatibility layer even if PipeWire drives the audio.

> The idea of reverting to pulse by default from my side was to give pipewire people a choice while not breaking existing setups.

Yes, they have a choice but the default is not what a PipeWire user is expecting.

Reply:
In fact most users have no idea what they are using, have running and configured properly. This is the initial feedback from the flatpak testers. They installed a linux distro and expect audio ;-).

And yeah, after the reading the above thread ... not much that can be done. So be it ...

For the "Revert" cherry-pick: Yes I indeed wondered, cause I made this Omega only, which is why I wonder you now have it "in some form" on a master rebase.

Please change to use PA's mainloop - and then we can call it a day. If it works for the bug reporters, you can then continue to backport, please.

For the record: I still don't like it - but I am a reality guy ;-) ...

Reply:
I appreciate the feedback :slightly_smiling_face: And I'm also not yet a 100 percent sold on this approach but so far I believe the benefits are worth the imperfections the solution has. I'll definitively keep this PR open for some time so more people can give their opinion.

Reply:
Revert is part of git. It does not break anything and revert is easy. I have seen PRs far too long standing somewhere without progress.